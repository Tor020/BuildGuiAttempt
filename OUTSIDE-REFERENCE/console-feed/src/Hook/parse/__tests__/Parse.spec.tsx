import * as _ from 'lodash'
import Parse from '..'

it('asserts values', () => {
  expect(Parse('assert', [2 > 1], 'assert-true')).toBe(false)
  expect(Parse('assert', [1 > 2], 'assert-false')).toMatchSnapshot(
    'assertion failed'
  )
})

it('counts numbers', () => {
  let final
  _.times(10, () => {
    final = Parse('count', ['count-10'])
  })

  expect(final && final.data[0]).toBe('count-10: 10')
})

it('profiles time', () => {
  Parse('time', ['timer-test'])

  setTimeout(() => {
    const result = Parse('timeEnd', ['timer-test'], 'timer-result')
    expect(result && +result.data[0].replace(/[^0-9]/g, '') > 100).toBeTruthy()
  }, 100)

  const failure = Parse('timeEnd', ['nonExistant'], 'timer-fail')
  expect(failure).toMatchSnapshot('non existant timer')
})

it('records errors', () => {
  const result = Parse('error', [new Error('one')], 'errors')

  expect(result && result.data[0]).toContain('Error: one')
})
