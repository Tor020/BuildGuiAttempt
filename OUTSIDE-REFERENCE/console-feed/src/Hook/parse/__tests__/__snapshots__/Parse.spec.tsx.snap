// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asserts values: assertion failed 1`] = `
Object {
  "data": Array [
    "Assertion failed:",
    "console.assert",
  ],
  "id": "assert-false",
  "method": "error",
}
`;

exports[`profiles time: non existant timer 1`] = `
Object {
  "data": Array [
    "Timer 'nonExistant' does not exist",
  ],
  "id": "timer-fail",
  "method": "warn",
}
`;
